src/pages/Facility/services/serviceRequests/ServiceRequestShow.tsx

```diff
@@ -73,20 +73,27 @@ export default function ServiceRequestShow({
     });
 
   const { mutate: createSpecimen } = useMutation({
-    mutationFn: mutate(specimenApi.createSpecimenFromDefinition, {
-      pathParams: {
-        facilityId,
-        serviceRequestId,
-      },
-    }),
+    mutationFn: async (variables: SpecimenFromDefinitionCreate) => {
+      const response = await mutate(specimenApi.createSpecimenFromDefinition, {
+        pathParams: {
+          facilityId,
+          serviceRequestId,
+        },
+      })(variables);
+
+      if (!response?.results?.length) {
+        throw new Error("No specimen created");
+      }
+      return response.results[0];
+    },
```

Comment:
use `onError` instead to raise errors.

stick with `mutationFn: mutate(...)` always